src/Protocol/Protocol18x.cpp

```diff
@@ -2548,10 +2548,33 @@ void cProtocol180::HandlePacketUpdateSign(cByteBuffer & a_ByteBuffer)
 		return;
 	}
 
+	std::map<std::string, std::string> SpecialCharacters;
+	SpecialCharacters["\\u0026"] = "&";
+	SpecialCharacters["\\u0027"] = "'";
+	SpecialCharacters["\\\""] = "\"";
+	SpecialCharacters["\\u003c"] = "<";
+	SpecialCharacters["\\u003e"] = ">";
+	SpecialCharacters["\\u003d"] = "=";
+
 	AString Lines[4];
 	for (int i = 0; i < 4; i++)
 	{
 		HANDLE_READ(a_ByteBuffer, ReadVarUTF8String, AString, Line);
+
+		// Buffer receives special characters in JSON code for sign messages, use SpecialCharacters map to parse them
+		size_t loc = 0;
+		for (const auto & itr : SpecialCharacters)
+		{
+			while (loc < Line.length())
```

Comment:
Minor comments: 
1. Putting `loc = 0` on top of the while loop and replacing `size_t loc = 0;` with `size_t loc;` is a bit cleaner in my opinion.
2. Using `Location` or `Index` is less vague than `loc`. Also, we use CamelCase, so `Loc`.